internal/ethapi/api.go

```diff
@@ -1028,6 +1027,7 @@ func DoCall(ctx context.Context, b Backend, args TransactionArgs, blockNrOrHash
 			args.KettleAddress = &acc
 		}
 
+		args.setDefaults(ctx, b)
```

Comment:
Estimate gas uses the `DoCall` function. We have to initialize the default values because the estimate gas is called without any gas limit or gas price.

Reply:
I think this GitHub comment would make a great code-comment.